src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/service/DataCleaner.java

```diff
@@ -68,12 +70,22 @@ public void add(ExtractionContainer container) {
     List<ExtractionContainer> deleted = new ArrayList<>(cleanupQueue);
     while (!cleanupQueue.isEmpty()) {
       ExtractionContainer container = cleanupQueue.poll();
-      log.info("Cleaning data at location {}", container.getExtractionUri());
+      log.debug("Cleaning data at location {}", container.getExtractionUri());
       deleteBucketAndContents(executorService, container);
+      deleteTable(container);
     }
     return deleted;
   }
 
+  private void deleteTable(ExtractionContainer container) {
+    List<PostExtractionAction> actions = container.getPostExtractionActions();
```

Comment:
Doesn't this mean we'll run this action twice - once now to delete the table and once later wherever we run the post extraction actions? Feels a bit wasteful to do it twice and to have this logic in two places. On the other hand doing something like this feels like it should belong in a cleaner. Do we even have other types of post extraction actions?

Reply:
Yes. I thought this too.
Yeah the `UpdateTableSchema` and `UpdatePartitionSchema` are both post extraction actions. All three are run after extracting the data.
The only way I could see to get around this is to run all post extraction actions _except_ the delete table, and then only run this in the cleaner at the end.